arduino/smartcar/smartcar.ino

```diff
@@ -1,9 +1,76 @@
 #include <Smartcar.h>
 
-void setup() {
-  // put your setup code here, to run once:
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+// const int frontPin = 0;
+// GP2Y0A02 sideFrontIR(arduinoRuntime,
+//                     frontPin); 
+
+const int backPin = 3;
+GP2Y0A02 sideBackIR(arduinoRuntime,
+                    backPin); 
+
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 2000;
+SR04 front(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+
+SimpleCar car(control);
+
+void setup()
+{
+    Serial.begin(9600);
+    Serial.setTimeout(200);
+}
+
+void loop()
+{
+    handleInput();
+    autoStop();
+    delay(100);
+   
 }
 
-void loop() {
-  // put your main code here, to run repeatedly:
+void handleInput()
+{
+    // handle serial input if there is any
+    if (Serial.available())
+    {
+        String input = Serial.readStringUntil('\n');
+        if (input.startsWith("m"))
+        {
+            int throttle = input.substring(1).toInt(); 
+            car.setSpeed(throttle);
+        }
+        else if (input.startsWith("t"))
+        {
+            int deg = input.substring(1).toInt();
+            car.setAngle(deg);
+        }
+        
+        
+    }
+   
 }
+
+void autoStop(){
+  
+  if(front.getDistance() < 1000 && front.getDistance() > 0){ 
+                car.setSpeed(-50);
+                delay(3000);
+                car.setSpeed(0);
```

Comment:
Why is this needed? Doesn't the car stop quick enough due to inertia?

Reply:
If we only have:
if(front.getDistance() < 100 && front.getDistance() > 0){ 
car.setSpeed (0);
}

then the car will stop when detecting an object. Furthermore, if the object is static (like a mountain or a tree) the distance won't change and the car speed will constantly be set to 0. So we added that the car is put in reverse for 3000 milliseconds and then breaks so that the front.getDistance() > 100, which will allow for new commands to be received.

Reply:
Aha I see. Not how I would do it (I would just not allow the car to move forward but it would be OK to move backward) but I guess you will eventually figure out a way on how to do this as the project evolves.

Generally having such large delays in your execution can prove very problematic.